src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -1097,127 +1081,6 @@ public static LivingSwapItemsEvent.Hands onLivingSwapHandItems(LivingEntity livi
         return event;
     }
 
-    public static void writeAdditionalLevelSaveData(WorldData worldData, CompoundTag levelTag) {
-        CompoundTag fmlData = new CompoundTag();
-        ListTag modList = new ListTag();
-        ModList.get().getMods().forEach(mi -> {
-            final CompoundTag mod = new CompoundTag();
-            mod.putString("ModId", mi.getModId());
-            mod.putString("ModVersion", MavenVersionStringHelper.artifactVersionToString(mi.getVersion()));
-            modList.add(mod);
-        });
-        fmlData.put("LoadingModList", modList);
-
-        CompoundTag registries = new CompoundTag();
-        fmlData.put("Registries", registries);
-        LOGGER.debug(WORLDPERSISTENCE, "Gathering id map for writing to world save {}", worldData.getLevelName());
-
-        for (Map.Entry<ResourceLocation, ForgeRegistry.Snapshot> e : RegistryManager.ACTIVE.takeSnapshot(true).entrySet()) {
-            registries.put(e.getKey().toString(), e.getValue().write());
-        }
-        LOGGER.debug(WORLDPERSISTENCE, "ID Map collection complete {}", worldData.getLevelName());
-        levelTag.put("fml", fmlData);
-    }
-
-    /**
-     * @param rootTag        Level data file contents.
-     * @param levelDirectory Level currently being loaded.
-     */
-    @ApiStatus.Internal
-    public static void readAdditionalLevelSaveData(CompoundTag rootTag, LevelStorageSource.LevelDirectory levelDirectory) {
-        CompoundTag tag = rootTag.getCompound("fml");
-        if (tag.contains("LoadingModList")) {
```

Comment:
We probably want to keep this, to make sure we keep firing `ModMismatchEvent`. (cc @RobotGryphon)

Reply:
Support for that event was restored. ;)